src/http-api.js

```diff
@@ -412,7 +412,7 @@ module.exports.MatrixHttpApi.prototype = {
         if (method == 'GET') {
             opts.qs = params;
         } else if (typeof params === "object") {
-            opts.form = params;
+            opts.json = params;
         } else if (typeof params === "string") {
             // Assume the caller has serialised the body to JSON
```

Comment:
Is it possible to remove this branch and all the places where I plopped `JSON.stringify(params)` in `idServerRequest` call sites? Your approach with `opts.json` seems much more sane, so it'd be nice to just do that everywhere...

Reply:
I can do that in a different PR, I'd rather not make this diff any harder to read if I can avoid it